File: advent2023/haskell/day1.hs
import System.IO
import Data.List
import Data.List.Split

isNum :: Char -> Bool
isNum x = x `elem` ['1'..'9']

findLeast :: String -> Char
findLeast [] = '0'
findLeast (x:xs) = if isNum x then x else findLeast xs

findMost :: String -> Char
findMost = findLeast . reverse 

lineCalc :: String -> String
lineCalc x = 
  let tens = findLeast x
      ones = findMost x
  in [tens] ++ [ones]

collectLines :: String -> [String]
collectLines = splitOn $ "\n"

main :: IO()
main = do
  input <- openFile "sample.txt" ReadMode
  raw_data <- hGetContents input
  putStrLn(show(sum(map (\x-> read x::Int) [lineCalc x| x <- collectLines raw_data])))

File: advent2018/day1.py
raw_data = open('sample.txt', 'r').read()
reached = set([0])
cur = 0
while True:
    for line in raw_data.split('\n')[:-1]:
        if line[0] == '+':
            cur += int(line[1:])
        else:
            cur -= int(line[1:])
        if cur not in reached:
            reached.add(cur)
        else:
            print('found')
            print(cur)
            exit()
print(cur)
